src/Containers/Microsoft.NET.Build.Containers/ImageIndexGenerator.cs

```diff
@@ -0,0 +1,128 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Text.Json;
+using System.Text.Json.Nodes;
+using Microsoft.NET.Build.Containers.Resources;
+using Microsoft.NET.Build.Containers.Tasks;
+
+namespace Microsoft.NET.Build.Containers;
+
+internal struct ImageInfo
+{
+    internal string Config { get; init; }
+    internal string ManifestDigest { get; init; }
+    internal string Manifest { get; init; }
+    internal string ManifestMediaType { get; init; }
+}
+
+internal static class ImageIndexGenerator
+{
+    /// <summary>
+    /// Generates an image index from the given images.
+    /// </summary>
+    /// <param name="imageInfos"></param>
+    /// <returns>Returns json string of image index and image index mediaType.</returns>
+    /// <exception cref="ArgumentException"></exception>
+    /// <exception cref="NotSupportedException"></exception>
+    internal static (string, string) GenerateImageIndex(ImageInfo[] imageInfos)
+    {
+        if (imageInfos.Length == 0)
+        {
+            throw new ArgumentException(string.Format(Strings.ImagesEmpty));
+        }
+
+        string firstManifestMediaType = imageInfos[0].ManifestMediaType;
+
+        if (firstManifestMediaType == SchemaTypes.DockerManifestV2)
```

Comment:
How likely are we to support more manifest types in the future? (I don't know if they exist and/or if they are reasonable to be supported)
Probably not too relevant right now, but switching as future proofing might be reasonable here.

Reply:
Not likely. These are the ones that exist today, to my knowledge

Reply:
that is also the reason for not going with "switch" - to emphasize there likely won't be others